In [1]:
import pandas as pd
# import numpy as np
# import matplotlib.pyplot as plt
# import math as math
%config Completer.use_jedi = False

## Step 1: Discernment of the data set

In [2]:
airTrafficDataFrame = pd.read_csv('./data files/Air_Traffic_Passenger_Statistics.csv',
                                   low_memory=False)
airTrafficDataFrame

Activity Period Operating Airline Operating Airline IATA Code  \
0               200507      ATA Airlines                          TZ   
1               200507      ATA Airlines                          TZ   
2               200507      ATA Airlines                          TZ   
3               200507        Air Canada                          AC   
4               200507        Air Canada                          AC   
...                ...               ...                         ...   
18393           201803  Volaris Airlines                          Y4   
18394           201803  Volaris Airlines                          Y4   
18395           201803           WOW Air                          WW   
18396           201803           WOW Air                          WW   
18397           201803           WOW Air                          WW   

      Published Airline Published Airline IATA Code    GEO Summary GEO Region  \
0          ATA Airlines                          TZ       Domestic         US   
1          ATA Airlines                          TZ       Domestic         US   
2          ATA Airlines                          TZ       Domestic         US   
3            Air Canada                          AC  International     Canada   
4            Air Canada                          AC  International     Canada   
...                 ...                         ...            ...        ...   
18393  Volaris Airlines                          Y4  International     Mexico   
18394  Volaris Airlines                          Y4  International     Mexico   
18395           WOW Air                          WW  International     Europe   
18396           WOW Air                          WW  International     Europe   
18397           WOW Air                          WW  International     Europe   

      Activity Type Code Price Category Code       Terminal Boarding Area  \
0               Deplaned            Low Fare     Terminal 1             B   
1               Enplaned            Low Fare     Terminal 1             B   
2         Thru / Transit            Low Fare     Terminal 1             B   
3               Deplaned               Other     Terminal 1             B   
4               Enplaned               Other     Terminal 1             B   
...                  ...                 ...            ...           ...   
18393           Deplaned            Low Fare  International             A   
18394           Enplaned            Low Fare  International             A   
18395           Enplaned               Other  International             A   
18396           Deplaned               Other  International             G   
18397           Enplaned               Other  International             G   

       Passenger Count  
0                27271  
1                29131  
2                 5415  
3                35156  
4                34090  
...                ...  
18393             4347  
18394             4228  
18395             4292  
18396             2699  
18397               10  

[18398 rows x 12 columns]

### Get information on all the columns

In [3]:
airTrafficDataFrame.info(memory_usage='deep')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18398 entries, 0 to 18397
Data columns (total 12 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   Activity Period              18398 non-null  int64 
 1   Operating Airline            18398 non-null  object
 2   Operating Airline IATA Code  18335 non-null  object
 3   Published Airline            18398 non-null  object
 4   Published Airline IATA Code  18335 non-null  object
 5   GEO Summary                  18398 non-null  object
 6   GEO Region                   18398 non-null  object
 7   Activity Type Code           18398 non-null  object
 8   Price Category Code          18398 non-null  object
 9   Terminal                     18398 non-null  object
 10  Boarding Area                18398 non-null  object
 11  Passenger Count              18398 non-null  int64 
dtypes: int64(2), object(10)
memory usage: 11.7 MB


^^^

There are 12 columns and 18398 rows. Data deficiencies occur in columns **'Operating Airline IATA Code'** and **'Published Airline IATA Code'**. The number of nan values in these two columns is the same.

### Get information on values' distributions and counts

In [4]:
columnNames = list(airTrafficDataFrame.columns)

for name in columnNames:
    print('Values in %s are unique: %s' % (name, airTrafficDataFrame[name].is_unique))

Values in Activity Period are unique: False
Values in Operating Airline are unique: False
Values in Operating Airline IATA Code are unique: False
Values in Published Airline are unique: False
Values in Published Airline IATA Code are unique: False
Values in GEO Summary are unique: False
Values in GEO Region are unique: False
Values in Activity Type Code are unique: False
Values in Price Category Code are unique: False
Values in Terminal are unique: False
Values in Boarding Area are unique: False
Values in Passenger Count are unique: False


^^^

There are no columns with unique values.

In [5]:
airTrafficDataFrame['Activity Period'].value_counts()

Activity Period
201708    156
201709    154
201710    153
201706    153
201707    151
         ... 
201111    110
201407    109
200602    108
201303    108
201301    107
Name: count, Length: 153, dtype: int64

In [6]:
airTrafficDataFrame['GEO Summary'].value_counts()

GEO Summary
International    11563
Domestic          6835
Name: count, dtype: int64

In [7]:
airTrafficDataFrame['Price Category Code'].value_counts()

Price Category Code
Other       16075
Low Fare     2323
Name: count, dtype: int64

In [8]:
airTrafficDataFrame['Terminal'].value_counts()

Terminal
International    11609
Terminal 1        3564
Terminal 3        2648
Terminal 2         550
Other               27
Name: count, dtype: int64

In [9]:
airTrafficDataFrame['Passenger Count'].value_counts()

Passenger Count
1        15
10       12
2        12
4        10
40        9
         ..
9160      1
8759      1
30857     1
30230     1
2699      1
Name: count, Length: 13708, dtype: int64

^^^

Get value counts of some colums.

### Get information about NaN (null) values

In [10]:
nullValuesOperatingCode = airTrafficDataFrame['Operating Airline IATA Code'].isnull()

nullValuesPublishedCode = airTrafficDataFrame['Published Airline IATA Code'].isnull()

In [11]:
df1 = airTrafficDataFrame[nullValuesOperatingCode]
df1

Activity Period     Operating Airline Operating Airline IATA Code  \
148             200508        Boeing Company                         NaN   
5715            201103             Servisair                         NaN   
6073            201103             Servisair                         NaN   
6289            201110             Servisair                         NaN   
6812            201005             Servisair                         NaN   
...                ...                   ...                         ...   
16868           201705  Trego Dugan Aviation                         NaN   
16869           201705  Trego Dugan Aviation                         NaN   
17631           201710         Swissport USA                         NaN   
17632           201710         Swissport USA                         NaN   
18062           201801         Swissport USA                         NaN   

          Published Airline Published Airline IATA Code    GEO Summary  \
148          Boeing Company                         NaN       Domestic   
5715              Servisair                         NaN       Domestic   
6073              Servisair                         NaN       Domestic   
6289              Servisair                         NaN       Domestic   
6812              Servisair                         NaN       Domestic   
...                     ...                         ...            ...   
16868  Trego Dugan Aviation                         NaN       Domestic   
16869  Trego Dugan Aviation                         NaN       Domestic   
17631         Swissport USA                         NaN  International   
17632         Swissport USA                         NaN  International   
18062         Swissport USA                         NaN       Domestic   

      GEO Region Activity Type Code Price Category Code       Terminal  \
148           US           Deplaned               Other          Other   
5715          US           Deplaned            Low Fare  International   
6073          US           Enplaned            Low Fare  International   
6289          US           Enplaned            Low Fare  International   
6812          US           Deplaned            Low Fare  International   
...          ...                ...                 ...            ...   
16868         US           Deplaned            Low Fare  International   
16869         US           Enplaned            Low Fare  International   
17631     Europe           Deplaned               Other  International   
17632     Europe           Enplaned               Other  International   
18062         US           Deplaned               Other  International   

      Boarding Area  Passenger Count  
148           Other               18  
5715              A               64  
6073              A               64  
6289              A               48  
6812              G               73  
...             ...              ...  
16868             A              108  
16869             A              304  
17631             A               73  
17632             A               73  
18062             A               98  

[63 rows x 12 columns]

In [12]:
df2 = airTrafficDataFrame[nullValuesPublishedCode]
df2

Activity Period     Operating Airline Operating Airline IATA Code  \
148             200508        Boeing Company                         NaN   
5715            201103             Servisair                         NaN   
6073            201103             Servisair                         NaN   
6289            201110             Servisair                         NaN   
6812            201005             Servisair                         NaN   
...                ...                   ...                         ...   
16868           201705  Trego Dugan Aviation                         NaN   
16869           201705  Trego Dugan Aviation                         NaN   
17631           201710         Swissport USA                         NaN   
17632           201710         Swissport USA                         NaN   
18062           201801         Swissport USA                         NaN   

          Published Airline Published Airline IATA Code    GEO Summary  \
148          Boeing Company                         NaN       Domestic   
5715              Servisair                         NaN       Domestic   
6073              Servisair                         NaN       Domestic   
6289              Servisair                         NaN       Domestic   
6812              Servisair                         NaN       Domestic   
...                     ...                         ...            ...   
16868  Trego Dugan Aviation                         NaN       Domestic   
16869  Trego Dugan Aviation                         NaN       Domestic   
17631         Swissport USA                         NaN  International   
17632         Swissport USA                         NaN  International   
18062         Swissport USA                         NaN       Domestic   

      GEO Region Activity Type Code Price Category Code       Terminal  \
148           US           Deplaned               Other          Other   
5715          US           Deplaned            Low Fare  International   
6073          US           Enplaned            Low Fare  International   
6289          US           Enplaned            Low Fare  International   
6812          US           Deplaned            Low Fare  International   
...          ...                ...                 ...            ...   
16868         US           Deplaned            Low Fare  International   
16869         US           Enplaned            Low Fare  International   
17631     Europe           Deplaned               Other  International   
17632     Europe           Enplaned               Other  International   
18062         US           Deplaned               Other  International   

      Boarding Area  Passenger Count  
148           Other               18  
5715              A               64  
6073              A               64  
6289              A               48  
6812              G               73  
...             ...              ...  
16868             A              108  
16869             A              304  
17631             A               73  
17632             A               73  
18062             A               98  

[63 rows x 12 columns]

In [13]:
df1.compare(df2)

Empty DataFrame
Columns: []
Index: []

^^^

The compare() method returned empty Data Frame, so null values are placed in in the same rows for both columns.